README.md

```diff
@@ -31,22 +31,26 @@ Each argument is prepended with "`--`".
 * The destination repository must accept HTTP PUT requests for storing maven artifacts and accept HTTP Basic authentication (successfully tested with Nexus3).
 
 ### Configuration
 The command-line arguments are as follows (arguments are prepended with "`--`"):
 
 Argument        | Mandatory | Description
-----------------|-----------|---------------
-source.root-url | Yes | The root URL from which the source repository should be scraped.
-source.user     | No  | The username used on the source repository (default: no authentication).
-source.password | No  | The password used on the source repository (default: no authentication).
-mirror-path     | No  | The path on the local file system to be used for mirroring repository content (default: `./mirror/`)
-target.root-url | Yes | The root URL where the content of the mirror repository should be uploaded.
-target.user     | No  | The username used on the target repository (default: no authentication).
-target.password | No  | The password used on the target repository (default: no authentication).
-actions         | No  | Comma-separated list of actions to be performed.<br/>Possible values:<ul><li>**`mirror`:** copy content from the source repository to the `mirror-path` on the local filesystem</li><li>**`publish`:** upload content from the `mirror-path` on the local filesystem to the target repository</li></ul>(default: `mirror,publish`)
+----------------|----------|---------------
+source.root-url          | Yes      | The root URL from which the source repository should be scraped.
+source.user              | No       | The username used on the source repository (default: no authentication).
+source.password          | No       | The password used on the source repository (default: no authentication).
+source.download-interval | No       | The interval between downloads, value in milliseconds, used to reduce load on source-server (default: 1000) 
+mirror-path              | No       | The path on the local file system to be used for mirroring repository content (default: `./mirror/`)
+target.root-url          | Yes      | The root URL where the content of the mirror repository should be uploaded.
+target.user              | No       | The username used on the target repository (default: no authentication).
+target.password          | No       | The password used on the target repository (default: no authentication).
+target.upload-interval   | No       | The interval between uploads, value in milliseconds, used to reduce load on target-server (default: 1000)
+target.concurrent-uploads| No       | Number of parallel uploads, parallelism is only supported on same folder, like foo/bar/artifact/1.0.0/ (default: 2)
+dry-run                  | No       | Boolean value (true or false) that disable write action, but run all read avaliable steps, useful when connections need to be checked without any change (default: false) 
```

Comment:
I thought about adding a `dry-run` option but chose the `actions` mechanism below because it is clearer what the tool actually does. I think the `dry-run` option now collides with the semantics of the `actions` option. Using `--actions=mirror` basically does the same (minus the http HEAD request, see below). 

Also I would expect a true "dry-run" to also **not** perform the mirror action, because that writes on the local filesystem and hence isn't really dry.

I'd prefer sticking to the `actions` option only. If performing a "connection-check" is your goal, I'd suggest adding a "check" action that does exactly this on the source and the target but nothing else.

Reply:
i agree with some of this, but, the target check isn't validated without starting to write without this dry-run action

Reply:
I'm not sure I understand what you mean here. My point is: rather than adding a `dry-run` parameter, capsule whatever is necessary for the connection-checking into a new `check` action.

I'd guess that a http HEAD request for the `target.root-url` would be enough to validate the target credentials isn't it?